src/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java

```diff
@@ -473,57 +447,107 @@ public boolean migrateAllLegacyPages(final User user, boolean respectFrontEndPer
     			}
     	return result;
     }
-    
+
     @Override
     public HTMLPageAsset migrateLegacyPage(HTMLPage legacyPage, User user, boolean respectFrontEndPermissions) throws Exception {
         Identifier legacyident=APILocator.getIdentifierAPI().find(legacyPage);
         VersionInfo vInfo=APILocator.getVersionableAPI().getVersionInfo(legacyident.getId());
-        
-        List<HTMLPageAsset> versions=new ArrayList<HTMLPageAsset>(); 
-        
+
         HTMLPage working=(HTMLPage) APILocator.getVersionableAPI().findWorkingVersion(legacyident, user, respectFrontEndPermissions);
-        HTMLPageAsset cworking=copyLegacyData(working, user, respectFrontEndPermissions), clive=null;
+        HTMLPageAsset cworking = migrateLegacyData(working, user, respectFrontEndPermissions), clive=null;
         if(vInfo.getLiveInode()!=null && !vInfo.getLiveInode().equals(vInfo.getWorkingInode())) {
             HTMLPage live=(HTMLPage) APILocator.getVersionableAPI().findLiveVersion(legacyident, user, respectFrontEndPermissions);
-            clive=copyLegacyData(live, user, respectFrontEndPermissions);
+            clive = migrateLegacyData(live, user, respectFrontEndPermissions);
         }
-        
+
         List<Permission> perms=null;
         if(!APILocator.getPermissionAPI().isInheritingPermissions(legacyPage)) {
             perms = APILocator.getPermissionAPI().getPermissions(legacyPage, true, true, true);
         }
-        
+
         List<MultiTree> multiTree = MultiTreeFactory.getMultiTree(working.getIdentifier());
```

Comment:
@agomez-dotcms let's leave this out for the moment since it is a part of a bigger discussion